### DISCUSSION TITLE
If condition on block or job

### DISCUSSION THREAD

**Discussion by joanofdart** [community/community#25980] [Actions]
<p>Hello!</p>
<p>I’m fairly new to the whole actions :slight_smile: and I’ve got some  questions.<br>
I want to stop a job from happening if the previous job didnt happen.</p>
<ul>
<li>How do I expose a variable from the previous job?</li>
<li>How do I check for the value of this variable?</li>
<li>Does a single if: blocks everything after it or just the next line?</li>
</ul>
<p>Thanks</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/joanofdart">@joanofdart</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>To answer your queries:</p>
<aside class="quote no-group" data-username="joanofdart" data-post="1" data-topic="120986" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 joanofdart:</div>
<blockquote>
<p>I want to stop a job from happening if the previous job didnt happen.</p>
</blockquote>
</aside>
<p>Jobs run in parallel by default. To run jobs sequentially, you can define dependencies on other jobs using the  <code>jobs.&lt;job_id&gt;.needs</code>  keyword. For example:</p>
<pre><code class="lang-auto">jobs:
  job1:
  job2:
    needs: job1
</code></pre>
<p>In this example,  <code>job1</code>  must complete successfully before  <code>job2</code>  begins, if job1 fails, the job2 will not begin.</p>
<aside class="quote no-group" data-username="joanofdart" data-post="1" data-topic="120986" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 joanofdart:</div>
<blockquote>
<ul>
<li>How do I expose a variable from the previous job?</li>
<li>How do I check for the value of this variable?</li>
</ul>
</blockquote>
</aside>
<p>Jobs are executed on individual VM, you can transfer the variable value to the next job which needs the previous one.<br>
‘set-ouput’ in previous job and get via <code>jobs.&lt;jobs_id&gt;.outputs.</code> in the 2nd job, please check the sample <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-6" rel="noopener nofollow ugc">here</a> for more details.</p>
<aside class="quote no-group" data-username="joanofdart" data-post="1" data-topic="120986" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 joanofdart:</div>
<blockquote>
<ul>
<li>Does a single if: blocks everything after it or just the next line?</li>
</ul>
</blockquote>
</aside>
<p>The if expression can be defined in step or job level, it only affects the step or job where it defines, not affect the next step or jobs.<br>
Please check the link for more details.<br>
Job level:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/52772761/181094402-536372f8-8c3e-4b78-91f9-beb29118a031.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions" target="_blank" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p>
step level:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/52772761/181094403-27751504-9793-4f00-a71d-7ec307433bb7.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions" target="_blank" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Hope it clear and helpful!</p>

**Comment by joanofdart** [comment]
<p>Hey <a class="mention" href="/weide-zhou">@weide-zhou</a></p>
<p>Thank you so much for replying.<br>
It was indeed clear and helpful.</p>
<p>Best wishes!</p>

**Comment by weide-zhou** [comment]
<p>Thanks <a class="mention" href="/joanofdart">@joanofdart</a> for your reply, glad to know it’s helpful.<br>
Could you please help to mark then the ticket can be archived, and it would help others who have same queries.</p>
<p>Cheers!</p>
